src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -391,6 +401,75 @@ protected virtual void Initialize()
             }
         }
 
+        // Scan the extensions directory and Load custom extension.
+        // $$$ - need to finalize this policy. Currently enabled via an AppSettting. 
+        private void LoadCustomExtensions(JObject hostConfig)
+        {
+            var bindingRoot = ConfigurationManager.AppSettings["BYOB_Path"];
+            if (!string.IsNullOrWhiteSpace(bindingRoot))
+            {
+
+                var config = this.ScriptConfig.HostConfig;
+
+                foreach (var dir in Directory.EnumerateDirectories(bindingRoot))
+                {
+                    foreach (var path in Directory.EnumerateFiles(dir, "*.dll"))
```

Comment:
We need to establish what he directory structure will be for extensions. Rather than a single directory dumping ground for all extensions, likely we want a directory per extension so different 3rd party dlls aren't conflicting?